ddtrace/contrib/graphql/patch.py

```diff
@@ -0,0 +1,275 @@
+    Note - graphql middlewares can not be a partial. It must be a class or a function.
+    """
+    pin = Pin.get_from(graphql)
+    if not pin or not pin.enabled():
+        return next_middleware(root, info, **args)
+
+    with pin.tracer.trace(
+        name="graphql.resolve",
+        resource=info.field_name,
+        span_type=SpanTypes.GRAPHQL,
+    ):
+        return next_middleware(root, info, **args)
+
+
+def _inject_trace_middleware_to_args(trace_middleware, args, kwargs):
+    # type: (Callable, Tuple, Dict) -> Tuple[Tuple, Dict]
+    """
+    Adds a trace middleware to graphql.execute(..., middleware, ...)
+    """
+    middlewares_arg = 8
+    if graphql_version >= (3, 2):
+        # middleware is the 10th argument graphql.execute(..) version 3.2+
+        middlewares_arg = 9
+
+    # get middlewares from args or kwargs
+    try:
+        middlewares = get_argument_value(args, kwargs, middlewares_arg, "middleware") or []
+        if isinstance(middlewares, MiddlewareManager):
+            # First we must get the middlewares iterable from the MiddlewareManager then append
+            # trace_middleware. For the trace_middleware to be called a new MiddlewareManager will
+            # need to initialized. This is handled in graphql.execute():
+            # https://github.com/graphql-python/graphql-core/blob/v3.2.1/src/graphql/execution/execute.py#L254
+            middlewares = middlewares.middlewares  # type: Iterable
+    except ArgumentError:
+        middlewares = []
+
+    # Note - graphql middlewares are called in reverse order
+    # add trace_middleware to the end of the list to wrap the execution of resolver and all middlewares
+    middlewares = list(middlewares) + [trace_middleware]
+
+    # update args and kwargs to contain trace_middleware
+    args, kwargs = set_argument_value(args, kwargs, middlewares_arg, "middleware", middlewares)
+    return args, kwargs
+
+
+def _get_source_str(obj):
+    # type: (Union[str, Source, Document]) -> str
+    """
+    Parses graphql Documents and Source objects to retrieve
+    the graphql source input for a request.
+    """
+    if isinstance(obj, str):
+        source_str = obj
+    elif isinstance(obj, Source):
+        source_str = obj.body
+    elif isinstance(obj, Document):
+        source_str = obj.loc.source.body
+    else:
+        source_str = ""
+    # remove new lines, tabs and extra whitespace from source_str
+    return re.sub(r"\s+", " ", source_str).strip()
+
+
+def _set_span_errors(errors, span):
+    # type: (List[GraphQLError], Span) -> None
+    if not errors:
+        # do nothing if the list of graphql errors is empty
+        return
+
+    error_msgs = "\n".join([stringify(error) for error in errors])
+    exc_type_str = "%s.%s" % (GraphQLError.__module__, GraphQLError.__name__)
+
+    span.error = 1
+    span._set_str_tag(ERROR_MSG, error_msgs)
+    span._set_str_tag(ERROR_TYPE, exc_type_str)
+    span._set_str_tag(ERROR_STACK, "")
```

Comment:
Let's see if it's possible to get a traceback, if not let's document why we can't here

Reply:
Maybe not set the tag?